### PR TITLE
feat(ci): add federated integ test for multi-ue to ci

### PR DESCRIPTION
**Pull request by orion-belt**
Signed-off-by: kharade <rohan.kharade@openairinterface.org>

## Summary

This PR adds Federated Integ Test  for multi-ue to CI. This task is a modification of [lte-integ-test.](https://github.com/magma/magma/blob/master/.github/workflows/lte-integ-test.yml#L3-L3)
## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/fabfile.py [hunk 1]
```diff
@@ -304,7 +304,8 @@ def federated_integ_test(
     execute(_make_integ_tests)
     sleep(20)
     execute(run_integ_tests, "federated_tests/s1aptests/test_attach_detach.py")
-
+    sleep(20) #ToDo: Remove individual test run and add in defs.mk
+    execute(run_integ_tests, "federated_tests/s1aptests/test_attach_detach_multi_ue.py")
```
Comment (uri200) [inline lte/gateway/fabfile.py]: Why does it have to run individually? Can t we do the same as LTE integ test where the test are run out of a list?

Comment (rohanrkharade) [reply lte/gateway/fabfile.py]: hi @uri200 
It is addressed now. I have created separate make option `fed_integ_test`.  Is it okay  ?

**Comment by voisey** [comment]
Please modify the documentation here https://docs.magmacore.org/docs/next/feg/s1ap_federated_test#s1ap-federated-integration-test if appropriate. Thanks!

**Comment by orion-belt** [comment]
> Please modify the documentation here https://docs.magmacore.org/docs/next/feg/s1ap_federated_test#s1ap-federated-integration-test if appropriate. Thanks!

Hi @voisey Will you check my [commit](https://github.com/magma/magma/pull/13243/commits/4a95764670a1139fa89ea79fc1380aea8758432b). I have updated the relevant mardown files.  Do I update any other files ?

File: docs/docusaurus/versioned_docs/version-1.7.0/feg/s1ap_federated_test.md [hunk 2]
```diff
@@ -190,7 +190,11 @@ vagrant ssh magma_test
 
 # inside vagrant vm
 cd magma/lte/gateway/python/integ_tests
-make integ_test TESTS=federated_tests/s1aptests/test_attach_detach.py
+## Run single test
```
Comment (voisey) [inline docs/docusaurus/versioned_docs/version-1.7.0/feg/s1ap_federated_test.md]: I don't think this change should be added to the v1.7 docs as this is a new test, i.e. one that wasn't used in v1.7, right?

File: docs/readmes/feg/s1ap_federated_test.md [hunk 3]
```diff
@@ -197,7 +197,11 @@ vagrant ssh magma_test
 
 # inside vagrant vm
 cd magma/lte/gateway/python/integ_tests
-make integ_test TESTS=federated_tests/s1aptests/test_attach_detach.py
+## Individual test(s):
```
Comment (voisey) [inline docs/readmes/feg/s1ap_federated_test.md]: Maybe something like `## Individual test(s), e.g.` could be better, to make it clear that `test_attach_detach.py` is just an example.

**Review by voisey — COMMENTED** [review]
Could you please also update `id` and `title` fields of the docs, as well as the title of the page, to reflect that these are integ tests and no longer just a single test? E.g. "S1AP Federated Integration Test" -> "S1AP Federated Integration Tests". It might also be nice to update the second paragraph on the page, which starts "Currently, this test..." to briefly mention this new test as well?

**Comment by orion-belt** [comment]
Hi @voisey 
I have addressed your comments [in the amendment](https://github.com/magma/magma/pull/13243/files). Will you please check now ?

File: docs/readmes/feg/s1ap_federated_test.md [hunk 4]
```diff
@@ -1,20 +1,24 @@
...
+id: s1ap_federated_tests
+title: S1AP Federated Integration Tests
 hide_title: true
 ---
 
-# S1AP Federated Integration Test
+# S1AP Federated Integration Tests
 
 The S1AP Integration Test only tests the AGW. The objective of the **S1AP
 Federated Integration Test** is to provide a test platform
```
Comment (voisey) [inline docs/readmes/feg/s1ap_federated_test.md]:
```suggestion
Federated Integration Tests** is to provide a test platform
```

File: docs/readmes/feg/s1ap_federated_test.md [hunk 5]
```diff
@@ -1,20 +1,24 @@
...
 to run Magma with all of its components. That is, an end-to-end test with all
 components of Magma: AGW, Orc8r and FeG.
 
 Currently, this test is in an experimental phase, so we are only testing the connectivity of
```
Comment (voisey) [inline docs/readmes/feg/s1ap_federated_test.md]: Maybe it should be "Currently, these tests are in..."?

File: docs/readmes/feg/s1ap_federated_test.md [hunk 6]
```diff
@@ -1,20 +1,24 @@
...
-AGW - Orc8r - FeG, and we are able to run an attach/detach test using the S1AP tester
+AGW - Orc8r - FeG, and we are able to run following tests using the S1AP tester
 and our mock HSS.
 
+
```
Comment (voisey) [inline docs/readmes/feg/s1ap_federated_test.md]: The Markdown lint check is complaining about the double blank lines here: https://github.com/magma/magma/runs/7296467637?check_suite_focus=true. Could you please remove one of them?

Note that in general you can check (and fix) this locally by following the instructions here https://github.com/magma/magma/wiki/Contributing-Documentation#precommit

File: docs/readmes/feg/s1ap_federated_test.md [hunk 7]
```diff
@@ -1,20 +1,24 @@
...
+# S1AP Federated Integration Tests
 
 The S1AP Integration Test only tests the AGW. The objective of the **S1AP
 Federated Integration Test** is to provide a test platform
 to run Magma with all of its components. That is, an end-to-end test with all
 components of Magma: AGW, Orc8r and FeG.
 
 Currently, this test is in an experimental phase, so we are only testing the connectivity of
-AGW - Orc8r - FeG, and we are able to run an attach/detach test using the S1AP tester
+AGW - Orc8r - FeG, and we are able to run following tests using the S1AP tester
```
Comment (voisey) [inline docs/readmes/feg/s1ap_federated_test.md]: Please insert a "the" after "run", so it's "run the following tests"

**Review by voisey — COMMENTED** [review]
Thanks for the update! I've added a few more comments now.

Also, now that the `id` and `title` have been updated, changes need to be made to a generated file, which the CI complains about here: https://github.com/magma/magma/runs/7296467848?check_suite_focus=true. Could you please run `make -C $MAGMA_ROOT/docs`, which should generate updates to `$MAGMA_ROOT/docs/docusaurus/i18n/en.json`, and then commit the changes to this file?

**Comment by orion-belt** [comment]
Hi @voisey 
Thanks for your comments.
I have addressed your comments [in the amendment](https://github.com/magma/magma/pull/13243/files).

File: lte/gateway/python/integ_tests/federated_tests/s1aptests/test_attach_detach_multi_ue.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
+"""
+Copyright 2020 The Magma Authors.
```
Comment (Neudrino) [inline lte/gateway/python/integ_tests/federated_tests/s1aptests/test_attach_detach_multi_ue.py]: Should be 2022.

Comment (rohanrkharade) [reply lte/gateway/python/integ_tests/federated_tests/s1aptests/test_attach_detach_multi_ue.py]: thanks @Neudrino It is addressed now.

**Comment by orion-belt** [comment]
@voisey 
So the [federated-integ-test](https://github.com/orion-belt/magma/runs/7325873701?check_suite_focus=true#logs) workflow runs fine now for multiple tests

**Review by voisey — APPROVED** [review]

File: lte/gateway/python/integ_tests/Makefile [hunk 9]
```diff
@@ -54,6 +54,16 @@ else
 	$(foreach test,$(EXTENDED_TESTS) $(PRECOMMIT_TESTS),$(call execute_test,$(test));)
 endif
 
+.PHONY: fed_integ_test
+fed_integ_test: $(PYTHON_BUILD)/setupinteg_env $(BIN)/nosetests
```
Comment (mpfirrmann) [inline lte/gateway/python/integ_tests/Makefile]: There was a switch from `nose` to `pytest` for testing.

```suggestion
fed_integ_test: $(PYTHON_BUILD)/setupinteg_env $(BIN)/pytest
```

**Review by mpfirrmann — CHANGES_REQUESTED** [review]

**Review by mpfirrmann — APPROVED** [review]

**Comment by voisey** [comment]
@orion-belt Just fyi, Markdown lint check has been fixed by https://github.com/magma/magma/pull/13339, so if you rebase, it should pass

**Comment by orion-belt** [comment]
Hi @uri200 @VinashakAnkitAman @ardzoht I have addressed comments of @voisey @Neudrino .
Will you please review changes again ?

**Review by VinashakAnkitAman — APPROVED** [review]
LGTM.. One mme test job is failing.. please rebase the PR

**Comment by orion-belt** [comment]
> LGTM.. One mme test job is failing.. please rebase the PR

Thanks @VinashakAnkitAman. I have [merged master](https://github.com/magma/magma/pull/13243/commits) in my branch

**Review by uri200 — APPROVED** [review]

**Review by ardzoht — APPROVED** [review]
lgtm

**Comment by Neudrino** [comment]
A clean rebase on master would be good here.

**Pull request merged**